clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java

```diff
@@ -50,6 +51,22 @@ public void testConfiguredInstances() {
         testInvalidInputs("org.apache.kafka.common.metrics.FakeMetricsReporter,");
     }
 
+
+    @Test
+    public void testEmptyList() {
```

Comment:
Can we add a test case which asserts the `ConfigException` if we pass `null` with the non-null validator? Also, a basic test case for the composite validator would be nice.